### DISCUSSION TITLE
PgCat HA Function

### DISCUSSION THREAD

**Discussion by TheOriginalGraLargeShrimpakaReaper** [postgresml/pgcat#728] [Q&A]
Has PgCat an HA Function?
Like VRRP?
My Goal is use a Primary PgCat Proxy / Connection Pooler Host in one Data Center and a Standby in the Fallback Data Center.
PostgreSQL runs with a 3 Node Patroni Cluster, the Primary and one Secondary Node are in the Main Data Center, the third is in the Fallback.

Of course I can setup keepalived and give PgCat a Virtual IP.
If the Primary PgCat Host Fails, the VIP make a Failover to the Secondary PgCat.
Some Architectures are setup for HAproxy, keepalived and Patroni.

But all Connections are going closed if the Primary PgCat Host fails.
So, if they are a better way, i'm open for it.
If this setup even functionally?

Greets

**Comment by cvhoangpt** [comment]
hi there, can pgcat read patroni api to check postgresql down node?
